src/components/app/details/appDetails/utils.tsx

```diff
@@ -213,27 +214,82 @@ export function getCalendarValue(startDateStr: string, endDateStr: string): stri
     return str;
 }
 
-export function getIframeSrc(appId: string | number, envId: string | number, environmentName: string, chartName: ChartTypes, newPodHash: string, calendarInputs, tab: AppMetricsTabType, isLegendRequired: boolean, statusCode?: string):string {
-    let rootUrl = process.env.REACT_APP_ORCHESTRATOR_ROOT.replace('/orchestrator', '');
-    // rootUrl = 'http://demo.devtron.info:32080';
-    let startTime: string = calendarInputs.startDate;
-    let endTime: string = calendarInputs.endDate;
-    let url = ``;
-    if (chartName !== 'status') {
-        url = `${rootUrl}/grafana/d-solo/devtron-app-metrics-`;
+export function isK8sVersionValid(k8sVersion: string): boolean {
+    if(!k8sVersion) return false;
+    try {
+        let version = (k8sVersion.split("v")[1]).split(".");
+        let versionNum = version.map(item => Number(item));
+        let sum = versionNum.reduce((sum, item) => {
+            return sum += item;
+        }, 0)
+        if (isNaN(sum)) return false;
+    } catch (error) {
+        return false
     }
-    else {
-        url = `${rootUrl}/grafana/d-solo/NnFpQOKGk/res_status_per_pod`;
+    return true;
+}
+
+export function isK8sVersion115OrBelow(k8sVersion: string): boolean {
+    let target = [1, 15, 1000];
+    let version: string[] = [];
+    let versionNum: number[] = [];
+    version = (k8sVersion.split("v")[1]).split(".");
+    versionNum = version.map(item => Number(item));
+
+    for (let i = 0; i < target.length; i++) {
+        if (versionNum[i] <= target[i]) {
+            return true;
+        }
     }
+    return false;
+}
+
+export interface AppInfo {
+    appId: string | number;
+    envId: string | number;
+    environmentName: string;
+    newPodHash: string,
+    k8sVersion: string;
+}
+
+export function getIframeSrc(appInfo: AppInfo, chartName: ChartTypes, calendarInputs, tab: AppMetricsTabType, isLegendRequired: boolean, statusCode?: string): string {
+    let baseURL = getGrafanaBaseURL(appInfo.k8sVersion, chartName);
+    let grafanaURL = addChartNameExtensionToBaseURL(baseURL, appInfo.k8sVersion, chartName, statusCode);
+    grafanaURL = addQueryParamToGrafanaURL(grafanaURL, appInfo.appId, appInfo.envId, appInfo.environmentName, chartName, appInfo.newPodHash, calendarInputs, tab, isLegendRequired, statusCode)
+    return grafanaURL;
+}
+
+export function getGrafanaBaseURL(k8sVersion, chartName): string {
+    let rootUrl = process.env.REACT_APP_ORCHESTRATOR_ROOT.replace('/orchestrator', '');
+    let url = '';
+    if (isK8sVersion115OrBelow(k8sVersion) && (chartName === 'cpu' || chartName === 'ram')) {
+        url = `${rootUrl}/grafana/d/devtron-app-metrics-`;
+    } else {
+        if (chartName !== 'status') {
```

Comment:
is this same as (chartName === 'cpu' || chartName === 'ram') ? if yes we should use same condition at both the places also we can simplify by setting

let url = `${rootUrl}/grafana/d-solo/NnFpQOKGk/res_status_per_pod`;
if (chartName === 'cpu' || chartName === 'ram') {
   if (isK8sVersion115OrBelow(k8sVersion)) {
            url = `${rootUrl}/grafana/d/devtron-app-metrics-`;
    } else {
            url = `${rootUrl}/grafana/d-solo/devtron-app-metrics-`;
    }
}

Reply:
There is a change in url. Following is the current logic:

let url = '/grafana/d-solo';
if (chartName === 'status') {
   url = `${url}/NnFpQOKGk/res_status_per_pod`;
 }
 else {
     url = `${url}/devtron-app-metrics-`;
}
return url;